src/enclave/rpc_sessions.h

```diff
@@ -354,7 +357,12 @@ namespace ccf
         }
         else
         {
-          auto ctx = std::make_unique<tls::Server>(certs[listen_interface_id]);
+          std::unique_ptr<tls::Context> ctx;
+          if (
+            per_listen_interface.endorsement.authority == Authority::UNSECURED)
+            ctx = std::make_unique<nontls::PlaintextServer>();
```

Comment:
Should we restrict which endpoint URI clients can trigger via this interface? Otherwise, they could be a couple of typos away (http:// instead of https:// and forgetting to type a port number, defaulting to `80`) from submitting their private payload to a plaintext endpoint?

Reply:
Yes, as discussed, we should. There is also an integrity issue with requests that are JWT authenticated, and virtually all responses other than receipts.

Reply:
Proposal coming up.

Reply:
See this commit (which is part of this PR): https://github.com/microsoft/CCF/pull/3945/commits/1c3cb5a208d390078c5a14126183426a327dbc2c

That's the simplest way I could think of, except I'm sure there is a way to get the local copy of the RPC interfaces instead of reading it transactionally. I'm not too familiar with the frontend infrastructure, so this may well contain errors/bugs/unintended consequences.